### PR TITLE
Run unit tests with Jest with --silent

### PR DESCRIPTION
**Pull request by ariya**
The console messages (mostly about React warning on deprecated methods) are useful when debugging a running application but they are atrocious, extremely noisy, and barely useful when running the unit tests.

In the CI, those console messages drown the more useful PASS/FAIL status of each individual test.

**Before**

Note how the log is extremely long (over 6,000 lines) comprises mostly React warning message:

image

**After**

As it should be, informing the PASS vs FAIL status of each test:

image

### CODE REVIEW

**Review by nemanjaglumac — APPROVED** [review]
Omg, yes please. 🚀

File: package.json [hunk 1]
```diff
@@ -167,9 +167,9 @@
     "lint-docs-links": "yarn && ./bin/verify-doc-links",
     "lint-yaml": "yamllint **/*.{yaml,yml} --ignore=node_modules/**/*.{yaml,yml}",
     "test": "yarn test-unit && yarn test-timezones && yarn test-cypress",
-    "test-unit": "yarn build-quick:cljs && jest --maxWorkers=2 --config jest.unit.conf.json",
+    "test-unit": "yarn build-quick:cljs && jest --silent --maxWorkers=2 --config jest.unit.conf.json",
     "test-unit-watch": "yarn test-unit --watch",
```
Comment (gusaiani) [inline package.json:171]: If we're going with this approach, I suggest adding it to the watch version too.

Comment (gusaiani) [reply package.json:171]: If you run `yarn test-unit-watch --silent someFilePath`, do you still get warnings?

I spent the last few hours with silent mode on...still getting the warnings ¯\_(¨)_/¯.

Comment (ariya) [reply package.json:171]: For me, Jest in watch mode doesn't give console message. There are `UnhandledPromiseRejectionWarning` but they are legitimate problems which need to be sorted out, not a mere console message.

**Pull request merged**